docs/GroupDefs.hpp

```diff
@@ -159,155 +164,485 @@
+%Parallel Components are called in random order and so it is not safe to have
+them depend on each other.
 
-and
-
+Each parallel component must also decide what to do in the different phases of
+the execution. This is controlled by an `execute_next_phase` function with
+signature:
 \code
-static void execute_next_global_actions(
+static void execute_next_phase(
     const typename metavariables::Phase next_phase,
     const Parallel::CProxy_ConstGlobalCache<metavariables>& global_cache);
 \endcode
+The `determine_next_phase` function in the Metavariables determines the next
+phase, after which the `execute_next_phase` function gets called. The
+`execute_next_phase` function determines what the %Parallel Component should do
+during the next phase. For example, it may simply call `perform_algorithm`, call
+a series of simple actions, perform a reduction over an Array, or not do
+anything at all. Note that `perform_algorithm` performs the same actions (the
+ones in `action_list`) no matter what Phase it is called in.
 
-The `initialize` function is called by the Main %Parallel Component when
-the execution starts and will typically call an explicit single Action
-to set up the initial state of the Algorithm, similar to what a constructor
-does for classes. The `initialize` function also receives arguments that
-are read from the input file and can the be used to initialize the %Parallel
-Component. For example, the value of an option could be distributed to all
-members of a %Parallel Component Array, or could be used to control the size
-of the %Parallel Component Array.  The `initialize` functions of different
-%Parallel Components are called in random order.
+An example of a singleton %Parallel Component is:
+\snippet Test_AlgorithmParallel.cpp singleton_parallel_component
 
-The `execute_next_global_actions` function gets run at the end of each
-%Parallel Phase and determines what the %Parallel Component should do
-during the next phase. For example, it may simply call `perform_algorithm`,
-call a series of single Actions, perform a reduction over an Array, or not
-do anything at all.  Note that `perform_algorithm` performs the same
-Actions (the ones in `action_list`) no matter what Phase it is called
-in.
+An example of an array %Parallel Component is:
+\snippet Test_AlgorithmParallel.cpp array_parallel_component
+Elements are inserted into the Array by using the Charm++ `insert` member
+function of the CProxy for the array. The `insert` function is documented in
+the Charm++ manual. In the above Array example `array_proxy` is a `CProxy` and
+so all the documentation for Charm++ array proxies applies. SpECTRE always
+creates empty Arrays with the constructor and requires users to insert however
+many elements they want and on which cores they want them to be placed. Note
+that load balancing calls may result in Array elements being moved.
+
+### %Actions
+
+For those familiar with Charm++, actions should be thought of as effectively
```

Comment:
Perhaps better start with a more approachable description of actions, instead of referring to the Charm++ concepts, since this will probably not help the reader at this point.

Reply:
The sentence is meant to allow anyone who knows about Charm++ to be able to get up to speed quicker. I could add something to the end of the sentence like "if not don't worry."

Reply:
My suggestion is to start more approachable, then give the reference to charm++ to help readers familiar with that.